tests/test_read_write/test_read_csv.py

```diff
@@ -0,0 +1,38 @@
+import pytest
```

Comment:
A lot of these tests are "repeating themselves". They have very similar code, that means that there's a use-case here to [parametrize the tests](https://calmcode.io/pytest/clean-params.html). This way the number of rows that you read in can be a parameter just like the path. In this I might also argue that the tests become a lot simpler if the file that you're using locally is the same as the file that you're reading online. For sake of simplicity I'd prefer to use `monopoly.csv` locally as well. 

It might also be good to confirm that the dictionaries that are created have the correct fields.

Reply:
I might also add some tests for the `fieldnames` argument.

Reply:
what tests do you have in mind for the fieldnames argument?

Reply:
In my mind the `fieldnames` parameter can help select which columns to select from the csv. That means that we should add all the columns if it is not given and only a subset if it is. 

This makes me wonder though how to deal with `csv` files that do not offer a header. 

Just to check, how do you think the implementation should handle; 

1. A csv file with a header on line 1 and no `fieldnames` passed. I assume we get all the columns? 
2. A csv file with a header on line 1. What might `fieldnames` do here? Rename the columns or subset? 
3. A csv file with no header. What might `fieldnames` do here? Give columns names? If so, can we subset? 

I'm wondering if we need two variables to deal with the no-header scenario.